activerecord/lib/active_record/connection_adapters/postgresql/oid/interval.rb

```diff
@@ -0,0 +1,42 @@
+require 'active_support/duration'
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module OID # :nodoc:
+        class Interval < Type::Value # :nodoc:
+          def type
+            :interval
+          end
+
+          def type_cast_from_database(value)
+            if value.kind_of? ::String
+              ::ActiveSupport::Duration.parse(value)
+            else
+              super
+            end
+          end
+
+          def type_cast_from_user(value)
+            type_cast_from_database(value)
+          rescue ::ActiveSupport::Duration::ISO8601ParsingError
+            value # Allow user to supply raw string values in another formats supported by PostgreSQL
```

Comment:
I think we should restrict this only to formats which we can parse for the time being. I don't like the idea of this type returning a string some of the time. We can always add support for more formats later.